core/src/ast_builder/assignment.rs

```diff
@@ -0,0 +1,55 @@
+use crate::{
+    ast::Assignment,
+    parse_sql::parse_sql_assignment,
+    result::{Error, Result},
+    translate::translate_assignment,
+};
+
+#[derive(Clone)]
+pub enum AssignmentNode {
+    Text(String),
+}
+
+impl From<&str> for AssignmentNode {
+    fn from(expr: &str) -> Self {
+        Self::Text(expr.to_owned())
+    }
+}
+
+impl TryFrom<AssignmentNode> for Assignment {
+    type Error = Error;
+
+    fn try_from(node: AssignmentNode) -> Result<Self> {
+        match node {
+            AssignmentNode::Text(expr) => {
+                let expr = parse_sql_assignment(expr).and_then(|op| translate_assignment(&op))?;
+                Ok(expr)
```

Comment:
[nit]

```suggestion
                parse_sql_assignment(expr).and_then(|op| translate_assignment(&op))
```

Reply:
This variable is an assignment, not an operator.
edit variable name

`op` to `assignment`